autowarefoundation/autoware
# [feature] Traffic Light Switch

## Status
**PRODUCTION / DEVELOPMENT**

## Description
This node tries to solve autowarefoundation/autoware_ai#106. The new feature use traffic signal obtained by image recognition and that recieved from external systems, for example AMS, and switch traffic signals safely. If either is red, it turns red. Also, timeout is set for external system information.

## Todos

## Steps to Test or Reproduce
1. Checkout this branch
2. Compile ./catkin_make
3. Launch traffic_light_recognition

        roslaunch road_wizard traffic_light_recognition.launch

4. Listen three traffic signals

        rostopic echo /camera_light_color  # color obtained by image recognition
        rostopic echo /ams_light_color  # color recieved from external systems
        rostopic echo /light_color  # color selected

5. If you publish some messages on `/camera_light_color` or `/ams_light_color`, you can see the change of color.

        rostopic pub /camera_light_color autoware_msgs/traffic_light "header:                                                                   
          seq: 0
          stamp:
            secs: 0
            nsecs: 0
          frame_id: ''
        traffic_light: 2"
